node.cpp

```diff
@@ -0,0 +1,49 @@
+
+#include <stdlib.h>
```

Comment:
I'd use C++ notation here: `#include <cstdlib>`, etc. and avoid mixing C and C++ libraries where possible. I'd default to C++ libraries.